### PR TITLE
Enhancement/editabletext limits

### LINKED ISSUE #1158: Handle limitations in EditableText
The `EditableText` component is used to edit labels in place. By default it allows infinite text and newlines, but that might not be desired behavior in some places. In the `PersonViewsPane` the title should not allow newlines and be limited to a certain length, but the description should have no limitations.

### PR DESCRIPTION
**Pull request by kristofferlarberg**
Resolves #1158. Adds possibility for text length limitation as well as prevents newlines in EditableText.

### CODE REVIEW

File: src/components/sections/people/PersonViewsPane.jsx
Comment (richardolsson) [inline src/components/sections/people/PersonViewsPane.jsx:177]: I think this property should be renamed to `multiline` or something like that (which is the inverse of what it does now, so logic will need to be slightly altered as well).

**Review by richardolsson — CHANGES_REQUESTED** [review]
Looking good! I'll let @niklasva82 review the main logic. I had just one comment on the naming of a property.

I'm also curious what happens when you paste text which contains newline characters. Will it allow that when "preventing" enter?

File: src/components/sections/people/PersonViewsPane.jsx
Comment (kristofferlarberg) [reply src/components/sections/people/PersonViewsPane.jsx:177]: Right, so basically the prop will be multiline={false}?

**Comment by kristofferlarberg** [comment]
> Looking good! I'll let @niklasva82 review the main logic. I had just one comment on the naming of a property.
> 
> I'm also curious what happens when you paste text which contains newline characters. Will it allow that when "preventing" enter?

Hmm, yes, at the moment all newlines are are removed onBlur, but this seems to be default? The const I added replaces newline with "" in order to make the field entirely empty when no text and hence show the placeholder text. Makes sense?

**Comment by niklasva82** [comment]
> > Looking good! I'll let @niklasva82 review the main logic. I had just one comment on the naming of a property.
> > I'm also curious what happens when you paste text which contains newline characters. Will it allow that when "preventing" enter?
> 
> Hmm, yes, at the moment all newlines are removed onBlur, but this seems to be default? The const I added replaces newline with "" in order to make the field entirely empty when no text and hence show the placeholder text. Makes sense?

This works and I think delving into the clipboard API is probably overkill at this point. At some point we should fix #1157 which is related to this and could be fixed by removing formatting from incoming pasted data, but I think this is outside the scope of this issue.

I do think the preventEnter prop should be renamed to multiline and inverted as @richardolsson suggested though before we merge this, it makes for a more easy to understand interface for this component.

**Comment by kristofferlarberg** [comment]
> I do think the preventEnter prop should be renamed to multiline and inverted as @richardolsson suggested though before we merge this, it makes for a more easy to understand interface for this component.

This is changed now and works!

**Comment by richardolsson** [comment]
Will you review again @niklasva82?

**Review by niklasva82 — APPROVED** [review]
Great work!

There is a bug where if you feed a newline and then start typing, you can type outside the EditableText window. But I think this was there before this change and I will file a bug report if there isn't one already.

**Review by richardolsson — APPROVED** [review]

**Pull request merged**
